test/Microsoft.Health.Fhir.R4.Tests.Integration/Features/ChangeFeed/SqlServerFhirResourceChangeCaptureTests.cs

```diff
@@ -0,0 +1,313 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Messages.Delete;
+using Microsoft.Health.Fhir.SqlServer.Features.ChangeFeed;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema;
+using Microsoft.Health.Fhir.Tests.Common;
+using Microsoft.Health.Fhir.Tests.Integration.Persistence;
+using Microsoft.Health.SqlServer;
+using Microsoft.Health.SqlServer.Configs;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Tests.Integration.Features.ChangeFeed
+{
+    /// <summary>
+    /// Integration tests for a resource change capture feature.
+    /// </summary>
+    public class SqlServerFhirResourceChangeCaptureTests
+    {
+        private const string LocalConnectionString = "server=(local);Integrated Security=true";
+        private IOptions<CoreFeatureConfiguration> _coreFeatureConfigOptions;
+
+        public SqlServerFhirResourceChangeCaptureTests()
+        {
+            _coreFeatureConfigOptions = Options.Create(new CoreFeatureConfiguration() { SupportsResourceChangeCapture = true });
+        }
+
+        /// <summary>
+        /// A basic smoke test verifying that the code can
+        /// insert and read resource changes after inserting a resource.
+        /// </summary>
+        [Fact]
+        public async Task GivenADatabaseSupportsResourceChangeCapture_WhenInsertingAResource_ThenResourceChangesShouldBeReturned()
+        {
+            FhirStorageTestsFixture fhirStorageTestsFixture = null;
+
+            try
+            {
+                for (int i = SchemaVersionConstants.SupportsResourceChangeCaptureSchemaVersion; i <= SchemaVersionConstants.Max; i++)
+                {
+                    string databaseName = $"FHIRRESOURCECHANGECAPTUREINSERTIONTEST_V{i}_{DateTimeOffset.UtcNow.ToUnixTimeSeconds()}";
+                    try
+                    {
+                        // this will either create the database or upgrade the schema.
+                        fhirStorageTestsFixture = new FhirStorageTestsFixture(new SqlServerFhirStorageTestsFixture(i, databaseName, _coreFeatureConfigOptions));
```

Comment:
Is it necessary to create a new test fixture for each of these tests? Could we instead keep the same database and check that the `resourceChanges.Count` increased by 1?

Reply:
Technically, it is possible if validating it on the same database for each test case. Currently, validating it on the multiple databases for schema versions that support a resource change capture feature.

Reply:
Unresolving this for now as it is related to [this discussion](https://github.com/microsoft/fhir-server/pull/2054#discussion_r668075412).

Reply:
Added a fixture class, and validated a resource change by the resource id.